### DISCUSSION TITLE
Anime Matrix key binding and automatic GPU mode changing

### DISCUSSION THREAD

**Discussion by salembegic** [seerge/g-helper#1865] [Ideas]
Hi everyone, I had a couple of ideas for new features that might be interesting, but I wanted to ask if they are something that would be useful to other people too, and what do you think would be the best way to implement them. 

1. The first thing I thought about is Anime Matrix key binding in the extra settings. The implementation should be pretty similar to any other bindings, but I feel like it would be nice to have a quick way to toggle through the brightness levels, or at least toggle the matrix on and off.

2. Second thing that came to mind is automatic switching of the GPU mode based on the performance profile. I'm not sure about everyone else, but I almost always use performance profiles and GPU modes in pairs (Silent → Eco, Balanced → Optimized, Turbo → Standard). It's not really a huge deal, but it would be nice if the app remembered the last pair that I used and then switched the GPU mode automatically when I changed the profile (For example if I used Silent and Eco, and then changed to Balanced and Optimized, next time I change to Silent it would automatically change back to Eco, of course you would be able to change it back if that's not what you want, and then it would remember for the next time). Of course there should probably be an option to turn this on and off as it might not be for everyone, but I feel like it would be a nice addition, and it would allow to quickly change the profiles with FN+F5 and not have to go to the app to adjust the GPU too. 

That's about it, feel free to add your opinions too, as I'm not sure if this is something that the majority wants.

**Comment by seerge** [comment]
@salembegic hello

1. Matrix brightness [hotkeys already exist](https://github.com/seerge/g-helper?tab=readme-ov-file#%EF%B8%8F-keybindings) : ``Fn + Shift + F7 / F8`` - Matrix brightness Down / Up 

2. Eco mode completely powers off your dGPU, there is no real reasons to do that if you are plugged on barrel charger (+ this is much more invasive operation than just switching power profile, so can't be easily linked there). If you want to automate Eco on battery and Standard when plugged it - it already exists, this is what Optimized do literally :)

If you really want hotkeys to Eco / Standard they also exist
``Ctrl + Shift + Alt + F14`` - Eco Mode
``Ctrl + Shift + Alt + F15`` - Standard Mode

**Reply by chrispoon007** [reply]
This might be a stupid question and I didn't want to start a new thread lol.

What are the f14/f15 keys for that shortcut? I tried ctrl+shift+alt+every key and also ctrl+shift+alt+fn+*key* and nothing changed the gpu mode to eco/standard mode. 

The reason I want GPU mode on Eco even when plugged-in is when I have fans off (for silent mode in class), the dGPU causes the laptop to heat up and eventually still turns on the fans (around 60-70c) even when idling. 
I guess I could force it to be Eco mode all the time until I turn it back on but a shortcut would make it easier or have silent mode be able to pair with eco.

I have a 2023 g14 btw so maybe other devices have a f14/f15 key? To be able to remap it would be great too.
Thanks so much for everything.
